salt/modules/debbuild.py

```diff
@@ -587,43 +599,131 @@ def make_repo(repodir,
+                            if stdout and SIGN_PROMPT_RE.search(stdout):
+                                # have the prompt for inputting the passphrase
+                                proc.sendline(phrase)
+                            else:
+                                times_looped += 1
+
+                            if times_looped > number_retries:
+                                raise SaltInvocationError(
+                                        'Attemping to sign file {0} failed, timed out after {1} loops'.format(abs_file, times_looped)
+                                 )
+                            # 0.125 is really too fast on some systems
+                            time.sleep(0.5)
+
+                        proc_exitstatus = proc.exitstatus
+                        if proc_exitstatus != 0:
+                            raise SaltInvocationError(
+                                 'Signing file {0} failed with proc.status {1}'.format(abs_file, proc_exitstatus)
+                                 )
+                    except salt.utils.vt.TerminalException as err:
+                        trace = traceback.format_exc()
+                        log.error(error_msg, err, trace)
+                    finally:
+                        proc.close(terminate=True, kill=True)
+
+        if use_passphrase:
             cmd = '/usr/lib/gnupg2/gpg-preset-passphrase --verbose --forget {0}'.format(local_fingerprint)
             __salt__['cmd.run'](cmd, runas=runas)
 
-            phrase = __salt__['pillar.get']('gpg_passphrase')
             cmd = '/usr/lib/gnupg2/gpg-preset-passphrase --verbose --preset --passphrase "{0}" {1}'.format(phrase, local_fingerprint)
             __salt__['cmd.run'](cmd, runas=runas)
 
-    repocfg_opts = _get_repo_options_env(env)
-    repoconfopts = os.path.join(repoconf, 'options')
-    with salt.utils.fopen(repoconfopts, 'w') as fow:
-        fow.write('{0}'.format(repocfg_opts))
-
     for debfile in os.listdir(repodir):
         abs_file = os.path.join(repodir, debfile)
         if debfile.endswith('.changes'):
             os.remove(abs_file)
 
         if debfile.endswith('.dsc'):
-            if local_keyid is not None:
-                cmd = 'debsign --re-sign -k {0} {1}'.format(keyid, os.path.join(repodir, abs_file))
+            if older_gnupg:
+                if local_keyid is not None:
+                    cmd = 'debsign --re-sign -k {0} {1}'.format(keyid, abs_file)
+                    __salt__['cmd.run'](cmd, cwd=repodir, use_vt=True)
+
+                cmd = 'reprepro --ignore=wrongdistribution --component=main -Vb . includedsc {0} {1}'.format(codename, abs_file)
                 __salt__['cmd.run'](cmd, cwd=repodir, use_vt=True)
-            cmd = 'reprepro --ignore=wrongdistribution --component=main -Vb . includedsc {0} {1}'.format(codename, abs_file)
-            __salt__['cmd.run'](cmd, cwd=repodir, use_vt=True)
+            else:
+                number_retries = 5
```

Comment:
Was planning to refactor, but did not have time to do so and test and submit a PR before vacation and tagging deadline.  Worked on the presumption working code is better than no code, esp. since needed to automate signing for Ubuntu 16.04